### ISSUE TITLE
No supported GPU(s) detected to run this container

### ISSUE THREAD

**Issue by HuaiyuanXu** [Yzichen/FlashOCC#3]
Thank you for your great work!

When I used the built image to create a container, the following error occurs:

WARNING: Detected NVIDIA NVIDIA GeForce RTX 4090 GPU, which is not yet supported in this version of the container
ERROR: No supported GPU(s) detected to run this container

Does this mean that NVIDIA TensorRT 22.07 (refer to"FROM nvcr.io/nvidia/tensorrt:22.07-py3" in the dockerfile) is incompatible to RTX 4090 GPU? Is FlashOCC available to run on other GPUs besides 3090?

![图片1](https://github.com/Yzichen/FlashOCC/assets/45089949/5d951b27-1e91-49ba-8444-f7af8dadd79f)

Many thanks!

**Comment by Yzichen** [comment]
I'm not sure about this one.

**Comment by HuaiyuanXu** [comment]
Many thanks for your reply! 
Could you share the pip list of your environment? This would help me to configure the running environment. Thanks.

**Comment by Yzichen** [comment]
(open-mmlab) zichen@lucky:~$ pip list
Package                       Version             Editable project location
----------------------------- ------------------- ---------------------------------------------------------------
absl-py                       2.0.0
actionlib                     1.14.0
addict                        2.4.0
aiofiles                      22.1.0
aiosqlite                     0.19.0
angles                        1.9.13
ansi2html                     1.8.0
anyio                         4.0.0
appdirs                       1.4.4
apptools                      5.2.1
argon2-cffi                   23.1.0
argon2-cffi-bindings          21.2.0
arrow                         1.2.3
asttokens                     2.4.0
astunparse                    1.6.3
async-lru                     2.0.4
asynctest                     0.13.0
attrs                         23.1.0
Babel                         2.12.1
backcall                      0.2.0
beautifulsoup4                4.12.2
bevdet-plugin                 0.0.0               /home/zichen/Documents/Project/BEV/Custom/BEVDet/projects
black                         23.9.1
bleach                        6.0.0
bondpy                        1.8.6
cachetools                    5.3.1
camera-calibration            1.17.0
camera-calibration-parsers    1.12.0
catkin                        0.8.10
ccimport                      0.4.2
certifi                       2023.7.22
cffi                          1.15.1
charset-normalizer            3.2.0
cityscapesScripts             2.2.2
click                         8.1.7
codecov                       2.1.13
colorama                      0.4.6
coloredlogs                   15.0.1
comm                          0.1.4
ConfigArgParse                1.7
configobj                     5.0.8
contourpy                     1.1.0
controller-manager            0.19.6
controller-manager-msgs       0.19.6
coverage                      7.3.1
cumm                          0.4.11
cv-bridge                     1.16.2
cycler                        0.11.0
Cython                        3.0.2
dash                          2.13.0
dash-core-components          2.0.0
dash-html-components          2.0.0
dash-table                    5.0.0
debugpy                       1.7.0
decorator                     5.1.1
defusedxml                    0.7.1
deprecation                   2.1.0
descartes                     1.1.0
diagnostic-analysis           1.11.0
diagnostic-common-diagnostics 1.11.0
diagnostic-updater            1.11.0
dynamic-reconfigure           1.7.3
einops                        0.6.1
entrypoints                   0.4
envisage                      7.0.3
exceptiongroup                1.1.3
executing                     1.2.0
fastbev-plugin                0.0.0               /home/zichen/Documents/Project/BEV/Custom/Fast-BEV/projects
fastjsonschema                2.18.0
fire                          0.5.0
flake8                        6.1.0
flash-attn                    0.2.2
flashocc-plugin               0.0.0               /home/zichen/Documents/Project/BEV/Custom/Temp/BEVDet/projects
Flask                         2.2.5
fonttools                     4.42.1
fqdn                          1.5.1
future                        0.18.3
fvcore                        0.1.5.post20221221
gast                          0.4.0
gazebo_plugins                2.9.2
gazebo_ros                    2.9.2
gencpp                        0.7.0
geneus                        3.0.0
genlisp                       0.4.18
genmsg                        0.6.0
gennodejs                     2.0.2
genpy                         0.6.15
google-auth                   2.23.0
google-auth-oauthlib          1.0.0
google-pasta                  0.2.0
grpcio                        1.58.0
h5py                          3.1.0
humanfriendly                 10.0
idna                          3.4
image-geometry                1.16.2
imagecorruptions              1.1.2
imageio                       2.31.3
importlib-metadata            6.8.0
importlib-resources           6.0.1
iniconfig                     2.0.0
interactive-markers           1.12.0
interrogate                   1.5.0
iopath                        0.1.10
ipykernel                     6.25.2
ipython                       8.12.2
ipython-genutils              0.2.0
ipywidgets                    8.1.0
isoduration                   20.11.0
isort                         4.3.21
itsdangerous                  2.1.2
jedi                          0.19.0
Jinja2                        3.1.2
joblib                        1.3.2
joint-state-publisher         1.15.1
joint-state-publisher-gui     1.15.1
json5                         0.9.14
jsonpointer                   2.4
jsonschema                    4.19.0
jsonschema-specifications     2023.7.1
jupyter                       1.0.0
jupyter_client                7.4.9
jupyter-console               6.6.3
jupyter_core                  5.3.1
jupyter-events                0.7.0
jupyter-lsp                   2.2.0
jupyter_packaging             0.12.3
jupyter_server                2.7.3
jupyter_server_fileid         0.9.0
jupyter_server_terminals      0.4.4
jupyter_server_ydoc           0.8.0
jupyter-ydoc                  0.2.5
jupyterlab                    3.6.6
jupyterlab-pygments           0.2.2
jupyterlab_server             2.24.0
jupyterlab-widgets            3.0.8
keras-nightly                 2.5.0.dev2021032900
Keras-Preprocessing           1.1.2
kiwisolver                    1.4.5
kwarray                       0.6.14
lark                          1.1.7
laser_geometry                1.6.7
lazy_loader                   0.3
llvmlite                      0.36.0
lyft-dataset-sdk              0.0.8
Markdown                      3.3.6
markdown-it-py                3.0.0
MarkupSafe                    2.1.3
matplotlib                    3.5.2
matplotlib-inline             0.1.6
mayavi                        4.8.1
mccabe                        0.7.0
mdurl                         0.1.2
message-filters               1.15.15
mistune                       3.0.1
mkl-fft                       1.3.1
mkl-random                    1.2.2
mkl-service                   2.4.0
mmcls                         1.0.0rc6
mmcv-full                     1.7.0
mmdet                         2.28.0              /home/zichen/Documents/Project/MMLab/mmdetection
mmdet3d                       1.0.0rc5            /home/zichen/Documents/Project/MMLab/mmdetection3d
mmsegmentation                0.29.1              /home/zichen/Documents/Project/MMLab/mmsegmentation
model-index                   0.1.11
modelindex                    0.0.2
mpmath                        1.3.0
mypy-extensions               1.0.0
nbclassic                     1.0.0
nbclient                      0.8.0
nbconvert                     7.8.0
nbformat                      5.7.0
nest-asyncio                  1.5.7
networkx                      2.2
ninja                         1.11.1
notebook                      6.5.6
notebook_shim                 0.2.3
numba                         0.53.0
numpy                         1.19.5
nuscenes-devkit               1.1.10
oauthlib                      3.2.0
olefile                       0.46
onnx                          1.7.0
onnxruntime                   1.15.1
open3d                        0.15.2
opencv-python                 4.8.0.76
opt-einsum                    3.3.0
ordered-set                   4.1.0
overrides                     7.4.0
packaging                     23.1
pandas                        1.4.4
pandocfilters                 1.5.0
parso                         0.8.3
pathspec                      0.11.2
pccm                          0.4.8
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        8.4.0
pip                           23.2.1
pkgutil_resolve_name          1.3.10
platformdirs                  3.10.0
plotly                        5.16.1
pluggy                        1.3.0
plyfile                       1.0.1
polarbev-plugin               0.0.0               /home/zichen/Documents/Project/BEV/Custom/PolarBEV/projects
portalocker                   2.7.0
prettytable                   3.9.0
prometheus-client             0.17.1
prompt-toolkit                3.0.39
protobuf                      3.20.1
psutil                        5.9.5
ptyprocess                    0.7.0
pure-eval                     0.2.2
py                            1.11.0
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pybind11                      2.11.1
pycocotools                   2.0.7
pycodestyle                   2.11.0
pycparser                     2.21
pyface                        8.0.0
pyflakes                      3.1.0
Pygments                      2.16.1
pyparsing                     3.1.1
pyquaternion                  0.9.9
pytest                        7.4.2
pytest-cov                    4.1.0
pytest-runner                 6.0.0
python-dateutil               2.8.2
python-json-logger            2.0.7
python-qt-binding             0.4.4
pytz                          2023.3.post1
PyWavelets                    1.4.1
PyYAML                        6.0.1
pyzmq                         24.0.1
qt-dotgraph                   0.4.2
qt-gui                        0.4.2
qt-gui-cpp                    0.4.2
qt-gui-py-common              0.4.2
qtconsole                     5.4.4
QtPy                          2.4.0
referencing                   0.30.2
renderocc                     1.0.0rc4
renderocc-plugin              1.0.0rc4            /home/zichen/Documents/Project/BEV/Official/RenderOcc
requests                      2.31.0
requests-oauthlib             1.3.1
resource_retriever            1.12.7
retrying                      1.3.4
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rich                          13.5.2
ros_numpy                     0.0.5
rosbag                        1.15.15
rosboost-cfg                  1.15.8
rosclean                      1.15.8
roscreate                     1.15.8
rosgraph                      1.15.15
roslaunch                     1.15.15
roslib                        1.15.8
roslint                       0.12.0
roslz4                        1.15.15
rosmake                       1.15.8
rosmaster                     1.15.15
rosmsg                        1.15.15
rosnode                       1.15.15
rosparam                      1.15.15
rospy                         1.15.15
rosservice                    1.15.15
rostest                       1.15.15
rostopic                      1.15.15
rosunit                       1.15.8
roswtf                        1.15.15
rpds-py                       0.10.2
rqt_action                    0.4.9
rqt_bag                       0.5.1
rqt_bag_plugins               0.5.1
rqt_console                   0.4.11
rqt_dep                       0.4.12
rqt_graph                     0.4.14
rqt_gui                       0.5.3
rqt_gui_py                    0.5.3
rqt_image_view                0.4.16
rqt_launch                    0.4.9
rqt_logger_level              0.4.11
rqt-moveit                    0.5.10
rqt_msg                       0.4.10
rqt_nav_view                  0.5.7
rqt_plot                      0.4.13
rqt_pose_view                 0.5.11
rqt_publisher                 0.4.10
rqt_py_common                 0.5.3
rqt_py_console                0.4.10
rqt-reconfigure               0.5.5
rqt-robot-dashboard           0.5.8
rqt-robot-monitor             0.5.14
rqt_robot_steering            0.5.12
rqt_runtime_monitor           0.5.9
rqt-rviz                      0.7.0
rqt_service_caller            0.4.10
rqt_shell                     0.4.11
rqt_srv                       0.4.9
rqt_tf_tree                   0.6.3
rqt_top                       0.4.10
rqt_topic                     0.4.13
rqt_web                       0.4.10
rsa                           4.8
rviz                          1.14.19
scikit-image                  0.19.3
scikit-learn                  1.3.0
scipy                         1.10.1
Send2Trash                    1.8.2
sensor-msgs                   1.13.1
setuptools                    59.5.0
Shapely                       1.8.5
six                           1.15.0
smach                         2.5.0
smach-ros                     2.5.0
smclib                        1.8.6
sniffio                       1.3.0
solofusion-plugin             0.0.0               /home/zichen/Documents/Project/BEV/Custom/SOLOFusion/projects
solofusionv1-plugin           0.0.0               /home/zichen/Documents/Project/BEV/Custom/SOLOFusionv1/projects
soupsieve                     2.5
spconv                        2.3.6
stack-data                    0.6.2
sympy                         1.12
tabulate                      0.9.0
tenacity                      8.2.3
tensorboard                   2.14.0
tensorboard-data-server       0.7.1
tensorboard-plugin-wit        1.8.1
termcolor                     2.3.0
terminado                     0.17.1
terminaltables                3.1.10
tf                            1.13.2
tf-conversions                1.13.2
tf2-geometry-msgs             0.7.6
tf2-kdl                       0.7.6
tf2-py                        0.7.6
tf2-ros                       0.7.6
threadpoolctl                 3.2.0
tifffile                      2023.7.10
tinycss2                      1.2.1
toml                          0.10.2
tomli                         2.0.1
tomlkit                       0.12.3
topic-tools                   1.15.15
torch                         1.9.1
torch-efficient-distloss      0.1.3
torch-scatter                 2.0.8
torchaudio                    0.9.0a0+a85b239
torchvision                   0.10.1
tornado                       6.3.3
tqdm                          4.66.1
traitlets                     5.9.0
traits                        6.4.3
traitsui                      8.0.0
trimesh                       2.35.39
typing                        3.7.4.3
typing_extensions             4.7.1
tzdata                        2023.3
ubelt                         1.3.3
uri-template                  1.3.0
urllib3                       1.26.16
vtk                           9.2.6
waymo-open-dataset-tf-2-5-0   1.4.1
wcwidth                       0.2.6
webcolors                     1.13
webencodings                  0.5.1
websocket-client              1.6.3
Werkzeug                      2.2.3
wheel                         0.38.4
widgetsnbextension            4.0.8
wrapt                         1.12.1
xacro                         1.14.14
xdoctest                      1.1.1
y-py                          0.6.2
yacs                          0.1.8
yapf                          0.40.1
ypy-websocket                 0.8.4
zipp                          3.17.0

**Comment by HuaiyuanXu** [comment]
Thanks for sharing.

I configured my environment according to BEVDet. (Link: https://github.com/HuangJunJie2017/BEVDet) (use its dev.3.0 branch)
But I encountered an issue using your training code:
![Weixin Screenshot_20231212195816](https://github.com/Yzichen/FlashOCC/assets/45089949/a3c6c524-051c-475a-801d-5d194630abe4)

Is there any guidance to fix this issue?

**Comment by Yzichen** [comment]
`cd projects`
`python setup.py develop`
